src/hooks/useGetTokenBalances.ts

```diff
@@ -42,28 +42,24 @@ const useBalanceCache = () => {
   const cacheRef = useRef<Record<string, BalanceCache>>({});
   const failedTokensRef = useRef<Set<string>>(new Set());
 
-  const getCached = useCallback(
-    (chain: Chain, wallet: WalletData, token: Token) => {
-      const key = `${chain}-${wallet.address}-${token.key}`;
-      const cached = cacheRef.current[key];
-      const now = Date.now();
+  const cacheKey = (wallet: string, token: Token): string => {
+    return `${token.chain}:${wallet}:${token.addressString}`;
+  };
 
-      if (cached && cached.lastUpdated > now - CACHE_DURATION_MS) {
-        return cached;
-      }
-      return null;
-    },
-    [],
-  );
+  const getCached = useCallback((wallet: WalletData, token: Token) => {
+    const key = cacheKey(wallet.address, token);
+    const cached = cacheRef.current[key];
+    const now = Date.now();
+
+    if (cached && cached.lastUpdated > now - CACHE_DURATION_MS) {
```

Comment:
Super long file, so I might have missed some logic/understanding.

If we check cached object staleness here before returning, is there is use case of the entire useEffect on line 80. Also that useEffect relies on intervals that are not guaranteed to run in the specified interval due to callback stacking, so wondering what happens if we just nuke out deleting things out of cache.

Reply:
I believe you are right - this could be removed.